inlong-sdk/dataproxy-sdk/src/main/java/org/apache/inlong/sdk/dataproxy/network/ClientMgr.java

```diff
@@ -572,29 +577,40 @@ public int compare(Map.Entry<HostInfo, Integer> o1, Map.Entry<HostInfo, Integer>
     }
 
     private void sendHeartBeat() {
-        for (HostInfo hostInfo : clientMap.keySet()) {
-            if (hostInfo == null) {
-                continue;
+        // all hbChannels need hb
+        for (Map.Entry<HostInfo, NettyClient> clientEntry : clientMapHB.entrySet()) {
+            if (clientEntry.getKey() != null && clientEntry.getValue() != null) {
+                sendHeartBeat(clientEntry.getKey(), clientEntry.getValue());
             }
-            NettyClient client = clientMap.get(hostInfo);
-            if (client == null) {
+        }
+
+        // only idle dataChannels need hb
+        for (Map.Entry<HostInfo, NettyClient> clientEntry : clientMapData.entrySet()) {
+            if (clientEntry.getKey() == null || clientEntry.getValue() == null) {
                 continue;
             }
-            try {
-                if (client.isActive()) {
-                    //logger.info("active host to send heartbeat! {}", entry.getKey().getHostName());
-                    EncodeObject encodeObject = new EncodeObject("heartbeat".getBytes(StandardCharsets.UTF_8),
-                            8, false, false, false, System.currentTimeMillis() / 1000, 1, "", "", "");
-                    if (configure.isNeedAuthentication()) {
-                        encodeObject.setAuth(configure.isNeedAuthentication(),
-                                configure.getUserName(), configure.getSecretKey());
-                    }
-                    client.write(encodeObject);
+            if (sender.isIdleClient(clientEntry.getValue())) {
+                sendHeartBeat(clientEntry.getKey(), clientEntry.getValue());
+            }
+        }
+
+    }
+
+    private void sendHeartBeat(HostInfo hostInfo, NettyClient client) {
+        try {
+            if (client.isActive()) {
```

Comment:
It is better to add a log when client is not active.

Reply:
done

Reply:
> It is better to add a log when client is not active.

done